### ISSUE TITLE
Can not calculate version

### ISSUE THREAD

**Issue by jaraco** [jaraco/jaraco.vcs#28]
Originally reported by: **Renier de Bruyn (BitBucket: [renierdbruyn](http://bitbucket.org/renierdbruyn), GitHub: [renierdbruyn](http://github.com/renierdbruyn))**

---

`hgtools` seems to break packages, when I try to install it trough pip i.e. `pip install https://github.com/renierdbruyn/calendra/tarball/master#calendra` or directly from your repo e.g. `pip install https://github.com/jaraco/calendra/tarball/master`

I got it to install by doing: `pip install git+git://github.com/renierdbruyn/calendra.git@master#egg=calendra` but this resets the version to 0.0.0 Why is this?

repo in question can be found [here](https://github.com/jaraco/calendra)

The stack trace:

``` python
Traceback (most recent call last):
  File "setup.py", line 61, in <module>
    setuptools.setup(**params)
  File "/usr/lib/python2.7/distutils/core.py", line 112, in setup
    _setup_distribution = dist = klass(attrs)
  File "/home/renierdebruyn/calendra/local/lib/python2.7/site-packages/setuptools/dist.py", line 264, in __init__
    _Distribution.__init__(self,attrs)
  File "/usr/lib/python2.7/distutils/dist.py", line 287, in __init__
    self.finalize_options()
  File "/home/renierdebruyn/calendra/local/lib/python2.7/site-packages/setuptools/dist.py", line 298, in finalize_options
    ep.load()(self, ep.name, value)
  File "build/bdist.linux-x86_64/egg/hgtools/plugins.py", line 113, in version_calc
  File "build/bdist.linux-x86_64/egg/hgtools/plugins.py", line 94, in calculate_version
  File "build/bdist.linux-x86_64/egg/hgtools/managers/base.py", line 41, in get_first_valid_manager
StopIteration

```

Am I doing something wrong?

---
- Bitbucket: https://bitbucket.org/jaraco/hgtools/issue/28

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

The StopIteration means that no hgtools Manager class was found that was suitable. Because you're doing a git checkout, that probably means that you don't have a late version of Git installed or in the path. hgtools 6.1 requires Git 1.7.10 or later. Older versions of hgtools probably won't help because they didn't work properly with Git for listing tags.

**Comment by jaraco** [comment]
_Original comment by_: **Gustavo Picon (BitBucket: [tabo](http://bitbucket.org/tabo), GitHub: [tabo](http://github.com/tabo))**

---

I'm affected by this issue too, and I do have git installed. The problem seems to be that hgtools is using setuptools' version parsing to compare version numbers, and this is what happens:

```
#!

>>> from pkg_resources import SetuptoolsVersion, SetuptoolsLegacyVersion
>>> SetuptoolsLegacyVersion('1.9.3 (Apple Git-50)') > SetuptoolsVersion('1.7.10')
False
```

So even while I have git 1.9.3 installed, hgtools discards it as if it was older than 1.7.10.

(The `StopIteration` exception could also be a little friendlier :)

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

@tabo The issue you're getting here I think is subtly different. SetuptoolsLegacyVersion was only very recently introduced in Setuptools 8, so hgtools probably needs to be updated to support that. I agreed the StopIteration is a poor experience.

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

Provide more information when the StopIteration happens. Ref #28.

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

Adapt parsing of git version output. Fixes failing test and fixes #28.

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

Add test capturing error on Mac OS X. Ref #28.

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

I've released the changes as 6.2. I believe with the improved error message, that's about as good as this issue will get, but if you have suggestions, please re-open or open another.
